src/coreservices.cpp

```diff
@@ -423,6 +423,29 @@ void CoreServices::initialize(QApplication* pApp) {
 
     m_pTouchShift = std::make_unique<ControlPushButton>(ConfigKey("[Controls]", "touch_shift"));
 
+    // The following UI controls must be created here so that controllers can bind to them
+    // on startup.
+    m_uiControls.clear();
+    const std::vector<ConfigKey> uiKeys = {
+            ConfigKey("[Master]", "skin_settings"),
+            ConfigKey("[Microphone]", "show_microphone"),
+            ConfigKey(VINYL_PREF_KEY, "show_vinylcontrol"),
+            ConfigKey("[PreviewDeck]", "show_previewdeck"),
+            ConfigKey("[Library]", "show_coverart"),
+            ConfigKey("[Master]", "maximize_library"),
+            ConfigKey("[Samplers]", "show_samplers"),
+            ConfigKey("[EffectRack1]", "show"),
+            ConfigKey("[Skin]", "show_4effectunits"),
+            ConfigKey("[Master]", "show_mixer"),
+    };
+
+    m_uiControls.reserve(uiKeys.size());
+    for (const auto& key : uiKeys) {
+        m_uiControls.emplace_back(std::make_unique<ControlPushButton>(
```

Comment:
persist = true already fetches the last value from the config.
The default value is used as a default if there is not yet anything stored.

https://github.com/mixxxdj/mixxx/blob/c7b4dbbb572930f8a5ae080aac71b99728c81966/src/control/control.cpp#L65

Reply:
ah got it.  Fixed.

Reply:
Hm, does that mean we need to pick the first-run defaults here?  Seems so